students/grant_dowell/Lesson_04/mailroom_part2.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+"""
+Mailroom Part 1
+Grant Dowell
+"""
+
+# Data structure requirements:
+#  - Name, history of donation amounts
+
+# Program Utilites
+#  1) Send Thank You
+#     - Prompt for full name
+#     - if
+#  2) Create a Report
+#  3) Quit
+
+def sortby_amt(entry):
+    return sum(entry[1])
+
+def generate_letter(name=None):
+    """ Builds a letter using the global database"""
+
+    template = "Dear {name},\n\n" + \
+               "Thank you for your most recent donation of " + \
+               "${last_donation}. We greatly appreciate it.\n\n" + \
+               " ~ The Treasurer"
+    donor_info = {'name':name, 'last_donation':db[name][-1]}
+    letter = template.format(**donor_info)
+    return letter
+
+def thanks():
+    """ Generates a Thank You message for a new donation, and adds donation
+        to database """
+
+    thank_you = ""
+    vld_name = False
+    while vld_name is False:
+        name = input("Please enter a Full Name or 'List': ")
+        if name.lower() == 'list':
+            for name in db.keys():
+                print(name)
+        elif name.lower() == 'quit':
+            break
+        else:
+            vld_name = True
+            for _ in db:    # Check for name in db
+                if name.lower == name.lower:
+                    break
+            else:   # If name not found in db, add it
+                db[name] = []
+    else:
+        print(name)
+        donation = input("Please enter a donation amount: ")
+        if donation.lower() != 'quit':  #Check if user is attempting to quit
+            db[name].append(float(donation))
+            thank_you = generate_letter(name)
+        print("\n" + thank_you + "\n")
+    return db, thank_you
```

Comment:
I don't believe you need this return data?

Reply:
Nope, deleted